tonic/src/transport/channel/tls.rs

```diff
@@ -15,6 +15,7 @@ pub struct ClientTlsConfig {
     cert: Option<Certificate>,
     identity: Option<Identity>,
     rustls_raw: Option<tokio_rustls::rustls::ClientConfig>,
+    _certs_validation: bool,
```

Comment:
I think we can remove the `_` prefix here?

Reply:
Problem is we only use that field when our new feature flag is enabled, resulting in a rustc warning when the feature flag is off.

Reply:
So I think the right option here is to just add `#[cfg(feature = "tls-dangerous")]` for each occurrence of this. So its impossible to even set if that feature flag is not set, what do you think?